### ISSUE TITLE
The spec should be clear about that mixins' interface can be implemented

### ISSUE THREAD

**Issue by yjbanov** [dart-lang/language#30]
I don't believe that the spec explicitly calls out the ability to `implement` a `mixin`. This should be added so that the relevant tools can implement and test this case sooner.

@lrhn @eernstg

**Comment by bwilkerson** [comment]
Which tools are the "relevant tools"?

A simple test indicates that analyzer currently supports this, so I'm trying to find out whether there's a bug in analyzer in some cases or whether you're running into problems in other tools.

**Comment by yjbanov** [comment]
> Which tools are the "relevant tools"?

CFE, VM, analyzer, DDC, dart2js come to mind.

> A simple test indicates that analyzer currently supports this, so I'm trying to find out whether there's a bug in analyzer in some cases or whether you're running into problems in other tools.

Practice shows a feature is rarely implemented well by accident, so I thought this should be formalized :)

**Comment by eernstg** [comment]
> I don't believe that the spec explicitly calls out the ability to `implement` a `mixin`.

Right, the [feature specification](https://github.com/dart-lang/language/blob/master/working/0006.%20Super-invocations%20in%20mixins/0007.%20Mixin%20declarations/lrhn-strawman.md) does not make it explicit that the `implements` clause of a class can contain mixins, but it gets very close:

> A mixin declaration defines an interface . The interface for this mixin
> declaration is equivalent to .. [an abstract class]

@lrhn, do you want to adjust the text to make this explicit?
